src/responder/common/responder_common.c

```diff
@@ -1024,12 +1094,22 @@ int activate_unix_sockets(struct resp_ctx *rctx,
 
         if (ret == numfds) {
             rctx->lfd = SD_LISTEN_FDS_START;
-            ret = sd_is_socket_unix(rctx->lfd, SOCK_STREAM, 1, NULL, 0);
+
+            ret = get_socket_path_in_use(tmp_ctx, rctx->lfd, &socket_path_in_use);
+            if (ret != EOK) {
+                DEBUG(SSSDBG_MINOR_FAILURE, "Unable to obtain socket path\n");
+            }
+
+            ret = sd_is_socket_unix(rctx->lfd, SOCK_STREAM, 1, socket_path_in_use, 0);
             if (ret < 0) {
                 DEBUG(SSSDBG_CRIT_FAILURE,
                       "Activated socket is not a UNIX listening socket\n");
                 ret = EIO;
                 goto done;
+            } else if (ret > 0){
+                DEBUG(SSSDBG_CONF_SETTINGS,
+                      "Warning: socket path defined in systemd unit (%s) and sssd.conf (%s) don't match\n",
+                      socket_path_in_use, rctx->sock_name);
```

Comment:
So first you get socket path for `rctx->lfd` and then you test that the socket path of `rctx->lfd` is socket path of `rctx->lfd`. IIUC this condition will never happen.